### PR TITLE
14329 fix mhr table filter bugs

### PR DESCRIPTION
**Pull request by ChasBoucher**
*Issue #:* /bcgov/entity#14329
*Description of changes:*

- Some minor fixes for filtering.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Review by doug-lovett — APPROVED** [review]
LGTM

File: ppr-ui/src/components/common/DatePicker.vue [hunk 1]
```diff
@@ -10,7 +10,7 @@
       </v-col>
       <v-col
         class="picker-title pl-4"
-        :class="{ 'picker-err': endDate === null && datePickerErr }"
+        :class="{ 'picker-err': endDate === null && datePickerErr && !disableEndDate }"
```
Comment (dimak1) [inline ppr-ui/src/components/common/DatePicker.vue]: Maybe we could rename the new prop to `enableEndDate` for clarity?

File: ppr-ui/src/components/tables/RegistrationTable.vue [hunk 2]
```diff
@@ -3,8 +3,9 @@
     <date-picker
       v-show="showDatePicker"
       ref="datePicker"
-      :setEndDate="submittedEndDate"
+      :setEndDate=null
```
Comment (dimak1) [inline ppr-ui/src/components/tables/RegistrationTable.vue]: Not to introduce more work, but maybe to clean up tech debt while at it: `setEndDate` and `setStartDate` are passing a _Date_ type object, but `endDate` and `startDate` are _String_ type props in `DatePicker`.

**Review by dimak1 — APPROVED** [review]
Approved, with few suggestions.

File: ppr-ui/src/components/tables/RegistrationTable.vue (hunk 2 above)
Comment (cameron-eyds) [reply ppr-ui/src/components/tables/RegistrationTable.vue]: Dima is 100% right here. 
Also, you shouldn't need to set `null` explicitly? Whats the default prop value on the component that it requires null here?

File: ppr-ui/src/enums/registrationTypes.ts [hunk 3]
```diff
@@ -168,7 +168,7 @@ export enum UIRegistrationClassTypes {
 export enum APIMhrDescriptionTypes {
   REGISTER_NEW_UNIT = 'REGISTER NEW UNIT',
   SALE_OR_GIFT = 'SALE / GIFT TRANSFER',
-  CONVERTED = '** CONVERTED **'
+  CONVERTED = 'RECORD CONVERSION'
```
Comment (cameron-eyds) [inline ppr-ui/src/enums/registrationTypes.ts:171]: This change happened on the API side?

File: ppr-ui/src/components/tables/RegistrationTable.vue (hunk 2 above)
Comment (cameron-eyds) [reply ppr-ui/src/components/tables/RegistrationTable.vue]: Also, does this impact PPR registrations? I know this table is a hybrid between the two.

**Review by cameron-eyds — COMMENTED** [review]
Looks good so far, just some minor questions!

File: ppr-ui/src/enums/registrationTypes.ts (hunk 3 above)
Comment (ChasBoucher) [reply ppr-ui/src/enums/registrationTypes.ts:171]: Yes.. I did test.

File: ppr-ui/src/components/tables/RegistrationTable.vue (hunk 2 above)
Comment (ChasBoucher) [reply ppr-ui/src/components/tables/RegistrationTable.vue]: I made the above change and changed the type in the datepicker. I tested both ppr and mhr and any other components that use the date picker and they all work.

File: ppr-ui/src/components/common/DatePicker.vue (hunk 1 above)
Comment (ChasBoucher) [reply ppr-ui/src/components/common/DatePicker.vue]: Good catch.. This code is not required after the change to allow for date range..

**Review by cameron-eyds — APPROVED** [review]
Looks good 👍

**Pull request merged**